src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQuerySnapshotPdao.java

```diff
@@ -301,23 +302,26 @@ public void createSnapshotWithLiveViews(Snapshot snapshot, Dataset dataset, Inst
     List<DatasetTable> tables = dataset.getTables();
 
     // create a snapshot table based on the live view data row ids
-    String liveViewTables =
-        createSnapshotTableFromLiveViews(
-            datasetBigQueryProject, tables, datasetBqDatasetName, filterBefore);
+    // Batch into smaller groups of tables to avoid BQ complexity errors
+    for (var tablesBatch : ListUtils.partition(tables, TABLE_UNION_BATCH_SIZE)) {
+      String liveViewTables =
+          createSnapshotTableFromLiveViews(
```

Comment:
Do we need to consider that a failure case can now occur where only some of the inserts are successful?  Should the set of insert operations now be included in a transaction so rollback can occur if one of the inserts fails?